lib/sync-DataSetModel.js

```diff
@@ -349,11 +354,16 @@ var self = {
     syncUtil.doLog(datasetClient.datasetId, 'silly', 'Now there are ' + datasetClient.pendingCallbacks.length + ' calbacks for datasetclient id =' + datasetClient.id, datasetClient);
   },
 
-  getClientHash: function (query_params, meta_data) {
-    var queryParamsHash = syncUtil.generateHash(query_params);
-    var metaDataHash = syncUtil.generateHash(meta_data);
+  getClientHash: function (query_params, meta_data, cb) {
+    syncUtil.generateHash(query_params, function(err, queryParamsHash) {
+      if (err) return cb && cb(err);
+
+      syncUtil.generateHash(meta_data, function(err, metaDataHash) {
```

Comment:
[Suggestion] Another improvement here would be to figure out where this hashes are used and instead of hashing twice maybe we can just append both together and generate single hash. Not sure how much changes this would require on client, but leaving this as suggestion.